docker build fails

```
git clone https://github.com/felt/tippecanoe.git
cd tippecanoe
docker build -t tc .
```
getting:
```
#10 15.21 projection.cpp: In function 'void lonlat2tile(double, double, int, long long int*, long long int*)':
#10 15.21 projection.cpp:25:32: error: 'fpclassify' was not declared in this scope
#10 15.21   int lat_class = fpclassify(lat);
#10 15.21                                 ^
#10 15.21 projection.cpp:25:32: note: suggested alternative:
#10 15.21 In file included from projection.cpp:5:0:
#10 15.21 /usr/include/c++/5/cmath:581:5: note:   'std::fpclassify'
#10 15.21      fpclassify(_Tp __x)
#10 15.21      ^
#10 15.21 projection.cpp: In function 'void epsg3857totile(double, double, int, long long int*, long long int*)':
#10 15.21 projection.cpp:77:30: error: 'fpclassify' was not declared in this scope
#10 15.21   int iy_class = fpclassify(iy);
#10 15.21                               ^
#10 15.21 projection.cpp:77:30: note: suggested alternative:
#10 15.21 In file included from projection.cpp:5:0:
#10 15.21 /usr/include/c++/5/cmath:581:5: note:   'std::fpclassify'
#10 15.21      fpclassify(_Tp __x)
#10 15.21      ^
#10 15.21 Makefile:74: recipe for target 'projection.o' failed
#10 15.21 make: *** [projection.o] Error 1
------
executor failed running [/bin/sh -c make   && make install]: exit code: 2
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,19 +1,20 @@
-# Start from ubuntu
-FROM ubuntu:16.04
+FROM ubuntu:22.04 AS tippacanoe-builder
 
-# Update repos and install dependencies
 RUN apt-get update \
-  && apt-get -y upgrade \
   && apt-get -y install build-essential libsqlite3-dev zlib1g-dev
 
-# Create a directory and copy in all files
-RUN mkdir -p /tmp/tippecanoe-src
-WORKDIR /tmp/tippecanoe-src
 COPY . /tmp/tippecanoe-src
+WORKDIR /tmp/tippecanoe-src
 
-# Build tippecanoe
-RUN make \
-  && make install
+RUN make 
 
-# Run the tests
 CMD make test
+
+# Using multistage build reduces the docker image size by alot by only copying the needed binaries
+FROM ubuntu:22.04
+RUN apt-get update \
+  && apt-get -y install libsqlite3-dev zlib1g-dev \
+  && rm -rf /var/lib/apt/lists/*
+COPY --from=tippacanoe-builder /tmp/tippecanoe-src/tippecanoe* /usr/local/bin/
+COPY --from=tippacanoe-builder /tmp/tippecanoe-src/tile-join /usr/local/bin/
+WORKDIR /app 

